Authentication/AuthenticationBootstrapper.swift

```diff
@@ -219,33 +235,29 @@ public class AuthenticationBootstrapper<User: UserType, SessionService: SessionS
 extension AuthenticationBootstrapper: LoginControllerTransitionDelegate {
     
     public func onSignup(controller: LoginController) {
-        let signupController = createSignupController()
-        if let navigationController = controller.navigationController {
-            navigationController.pushViewController(signupController, animated: true)
+        if _initialScreen == .Login {
+            let signupController = createSignupController()
+            controller.navigationController!.pushViewController(signupController, animated: true)
         } else {
-            _window.rootViewController = UINavigationController(rootViewController: signupController)
+            controller.navigationController!.popViewControllerAnimated(true)
         }
     }
     
     public func onRecoverPassword(controller: LoginController) {
         let recoverPasswordController = createRecoverPasswordController()
-        if let navigationController = controller.navigationController {
-            navigationController.pushViewController(recoverPasswordController, animated: true)
-        } else {
-            _window.rootViewController = UINavigationController(rootViewController: recoverPasswordController)
-        }
+        controller.navigationController!.pushViewController(recoverPasswordController, animated: true)
     }
     
 }
 
 extension AuthenticationBootstrapper: SignupControllerTransitionDelegate {
     
     public func onLogin(controller: SignupController) {
```

Comment:
What does this onLogin means? When the user logins or when he presses the button? If it is the second one, we should change its name